arangod/Network/Utils.cpp

```diff
@@ -215,94 +215,92 @@ int fuerteToArangoErrorCode(fuerte::Error err) {
+      return network::opResultFromBody(std::move(body), TRI_ERROR_INTERNAL);
   }
 }
 
 /// @brief Create Cluster Communication result for document
 OperationResult clusterResultDocument(arangodb::fuerte::StatusCode code,
                                       std::shared_ptr<VPackBuffer<uint8_t>> body,
-                                      OperationOptions const& options,
+                                      OperationOptions options,
                                       std::unordered_map<int, size_t> const& errorCounter) {
   switch (code) {
     case fuerte::StatusOK:
-      return OperationResult(Result(), std::move(body), options, errorCounter);
+      return OperationResult(Result(), std::move(body), std::move(options), errorCounter);
     case fuerte::StatusPreconditionFailed:
       return OperationResult(Result(TRI_ERROR_ARANGO_CONFLICT), std::move(body),
-                             options, errorCounter);
+                             std::move(options), errorCounter);
     case fuerte::StatusNotFound:
-      return network::opResultFromBody(body, TRI_ERROR_ARANGO_DOCUMENT_NOT_FOUND);
+      return network::opResultFromBody(std::move(body), TRI_ERROR_ARANGO_DOCUMENT_NOT_FOUND);
     default:
-      return network::opResultFromBody(body, TRI_ERROR_INTERNAL);
+      return network::opResultFromBody(std::move(body), TRI_ERROR_INTERNAL);
   }
 }
   
 /// @brief Create Cluster Communication result for modify
 OperationResult clusterResultModify(arangodb::fuerte::StatusCode code,
                                     std::shared_ptr<VPackBuffer<uint8_t>> body,
-                                    OperationOptions const& options,
+                                    OperationOptions options,
                                     std::unordered_map<int, size_t> const& errorCounter) {
   switch (code) {
     case fuerte::StatusAccepted:
     case fuerte::StatusCreated: {
-      OperationOptions options;
       options.waitForSync = (code == fuerte::StatusCreated);
-      return OperationResult(Result(), std::move(body), options, errorCounter);
+      return OperationResult(Result(), std::move(body), std::move(options), errorCounter);
     }
     case fuerte::StatusConflict:
-      return OperationResult(network::resultFromBody(body, TRI_ERROR_ARANGO_UNIQUE_CONSTRAINT_VIOLATED),
-                             body, options, errorCounter);
+      return OperationResult(network::resultFromBody(std::move(body), TRI_ERROR_ARANGO_UNIQUE_CONSTRAINT_VIOLATED),
+                             std::move(body), std::move(options), errorCounter);
     case fuerte::StatusPreconditionFailed:
-      return OperationResult(network::resultFromBody(body, TRI_ERROR_ARANGO_CONFLICT),
-                             body, options, errorCounter);
+      return OperationResult(network::resultFromBody(std::move(body), TRI_ERROR_ARANGO_CONFLICT),
+                             std::move(body), std::move(options), errorCounter);
```

Comment:
this statement moves from `body` twice. is it defined in which order the function call parameters are evaluated?